code/game/mecha/mech_bay.dm

```diff
@@ -165,6 +197,75 @@
 	else
 		return 0
 
+/obj/machinery/mech_bay_recharge_port/update_icon()
+	. = ..()
+	icon_state = "[initial(icon_state)][anchored? "":"-unwrenched"]"
+
+/obj/machinery/mech_bay_recharge_port/wrenchAnchor()
+	. = ..()
+	update_icon()
+	if(!anchored)
+		stop_charge()
+		delink_devices()
+
+/obj/machinery/mech_bay_recharge_port/examine(mob/user)
+	. = ..()
+	if(recharge_floor)
+		var/direction = get_dir_as_string(recharge_floor)
+		to_chat(user,"<span class='notice'>Linked to \the [recharge_floor] at the [uppertext(direction)].</span>")
+	if(recharge_console)
+		var/direction = get_dir_as_string(recharge_console)
+		to_chat(user,"<span class='notice'>Linked to \the [recharge_console] at the [uppertext(direction)].</span>")
+
+/obj/machinery/mech_bay_recharge_port/Destroy()
+	delink_devices()
+	..()
+
+/obj/machinery/mech_bay_recharge_port/proc/delink_devices()
+	if(recharge_floor)
+		recharge_floor.recharge_port = null
+		recharge_floor = null
+	if(recharge_console)
+		recharge_console.recharge_port = null
+		recharge_console = null
+
+/obj/machinery/mech_bay_recharge_port/proc/locate_and_link_station()
+	if(recharge_floor) // we already have a station
+		return 0
+	force_cardinal_dir()
+	for(var/obj/machinery/mech_bay_recharge_floor/potential_recharge_floor in get_step(src,dir))
+		if(!potential_recharge_floor.anchored)
+			continue
+		if(potential_recharge_floor.recharge_port) //it is already linked
+			continue
+		recharge_floor = potential_recharge_floor
+		recharge_floor.recharge_port = src
+		recharge_floor.recharge_console = recharge_console
+		recharge_console.recharge_floor = recharge_floor
+		return 1
+	return 0
+
+/obj/machinery/mech_bay_recharge_port/verb/rotate_cw()
+	set name = "Rotate (Clockwise)"
+	set category = "Object"
+	set src in oview(1)
+
+	if(src.anchored || usr:stat)
+		to_chat(usr, "<span class='warning'>It is fastened to the floor!</span>")
+		return 0
+	src.dir = turn(src.dir, -90)
+	return 1
+
+/obj/machinery/mech_bay_recharge_port/verb/rotate_ccw()
+	set name = "Rotate (Counter-Clockwise)"
+	set category = "Object"
+	set src in oview(1)
+
+	if(src.anchored || usr:stat)
```

Comment:
Instead of using the evil runtime search operator just check that usr is a mob then check the stat var on the mob directly.